src/main/java/net/mcreator/ui/procedure/RetvalProcedureSelector.java

```diff
@@ -0,0 +1,251 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.procedure;
+
+import net.mcreator.blockly.data.Dependency;
+import net.mcreator.element.ModElementType;
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.element.parts.procedure.RetvalProcedure;
+import net.mcreator.generator.GeneratorConfiguration;
+import net.mcreator.generator.GeneratorStats;
+import net.mcreator.io.net.analytics.AnalyticsConstants;
+import net.mcreator.java.JavaConventions;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.JEmptyBox;
+import net.mcreator.ui.component.util.ComboBoxFullWidthPopup;
+import net.mcreator.ui.component.util.ComponentUtils;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.help.IHelpContext;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.modgui.ModElementGUI;
+import net.mcreator.ui.validation.component.VTextField;
+import net.mcreator.ui.validation.optionpane.OptionPaneValidatior;
+import net.mcreator.ui.validation.optionpane.VOptionPane;
+import net.mcreator.ui.validation.validators.UniqueNameValidator;
+import net.mcreator.util.StringUtils;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.VariableType;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+public abstract class RetvalProcedureSelector<E, T extends RetvalProcedure<E>> extends AbstractProcedureSelector {
+
+	@Nullable private final JComponent fixedValue;
+
+	public RetvalProcedureSelector(VariableType returnType, @Nullable IHelpContext helpContext, MCreator mcreator,
+			String eventName, Side side, @Nullable JComponent fixedValue, int width,
+			Dependency... providedDependencies) {
+		super(mcreator, returnType, providedDependencies);
+
+		this.fixedValue = fixedValue;
+		this.defaultName = L10N.t("procedure.common.fixed");
+
+		setLayout(new BorderLayout(0, 0));
+
+		addMouseListener(new MouseAdapter() {
+			@Override public void mouseClicked(MouseEvent e) {
+				e.consume();
+			}
+		});
+
+		setBorder(BorderFactory.createEmptyBorder(0, 2, 0, 2));
+
+		setOpaque(true);
+		procedures.setBorder(BorderFactory.createLineBorder(returnType.getBlocklyColor()));
+		setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
+
+		procedures.setRenderer(new ConditionalComboBoxRenderer());
+		procedures.addPopupMenuListener(new ComboBoxFullWidthPopup());
+		procedures.addActionListener(e -> {
+			CBoxEntry selectedItem = procedures.getSelectedItem();
+			if (selectedItem != null) {
+				if (!selectedItem.correctDependencies) {
+					procedures.setSelectedItem(oldItem);
+				} else {
```

Comment:
You don't use brackets below, so you should try to keep consistency inside the same class. There are also other classes (in this PR) that would require to be more consistent towards brackets.

Reply:
I prefer for brackets to be there, but are not mandatory. Intellij handles codestyle for us with occasional reformats

Reply:
It was purely for the code style.

Reply:
I prefer brackets vs no brackets as it is more clear. No brackets is usually from old code. I may change code style formatted in the future to format with brackets where they are missing.